code/modules/reagents/chemistry/reagents/unique/eigenstasium.dm

```diff
@@ -0,0 +1,110 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//										EIGENSTASIUM
+///////////////////////////////////////////////////////////////////////////////////////////////////
+//eigenstate Chem
+//Teleports you to the creation location and back
+//OD teleports you randomly around the Station and gives you a status effect
+//The status effect slowly send you on a wild ride and replaces you with an alternative reality version of yourself unless you consume eigenstasium/bluespace dust.
+//During the process you get really hungry, then some of your items slowly start teleport around you,
+//then alternative versions of yourself are brought in from a different universe and they yell at you.
+//and finally you yourself get teleported to an alternative universe, and character your playing is replaced with said alternative
+//Spraying this on lockers allows you to create eigenlinked lockers - see the eigenstate subsystem for using this to create your own links
+
+/datum/reagent/eigenstate
+	name = "Eigenstasium"
+	description = "A strange mixture formed from a controlled reaction of bluespace with plasma, that causes localised eigenstate fluxuations within the patient"
+	taste_description = "wiggly cosmic dust."
+	color = "#5020F4"
+	overdose_threshold = 15
+	metabolization_rate = 1 * REAGENTS_METABOLISM
+	ph = 3.7
+	impure_chem = /datum/reagent/impurity/eigenswap
+	inverse_chem = null
+	failed_chem = /datum/reagent/bluespace //crashes out
+	chemical_flags = REAGENT_DEAD_PROCESS //So if you die with it in your body, you still get teleported back to the location as a corpse
+	data = list("location_created" = null, "creator" = null)//So we retain the target location and creator between reagent instances
+	///The creation point assigned during the reaction
+	var/turf/location_created
+	///The mob that initial teleport is linked to
+	var/mob/living/carbon/creator
```

Comment:
Either make this a weakref or hook onto its deletion, otherwise you'll get hard deletes.

Reply:
The whole blood thing isn't really needed - it was from feedback that changed so it's probably just easier to not have that behavour. I got rid of it, but at least I know what to do if I ever want a name to a chemist.